src/Tribe/Aggregator/REST/V1/Endpoints/Batch.php

```diff
@@ -0,0 +1,176 @@
+
+		/** @var Tribe__Events__Aggregator__Record__Abstract $record */
+		$record = $records->get_by_import_id( $request['import_id'], array( 'post_status' => 'any' ) );
+
+		if ( empty( $record->meta['post_status'] ) ) {
+			$record->update_meta( 'post_status', tribe( 'events-aggregator.settings' )->default_post_status( $record->meta['origin'] ) );
+		}
+
+		// let's make sure it's a nested object
+		$items = json_decode( json_encode( $request['events'] ) );
+
+		$is_last_batch = (int) $request['status']['batch']['total'] === ( (int) $request['status']['batch']['done'] + 1 );
+
+		/** @var Tribe__Events__Aggregator__Record__Activity $activity */
+		$activity = $record->insert_posts( $items );
+
+		$is_success  = $activity->get_last_status() === Tribe__Events__Aggregator__Record__Activity::STATUS_SUCCESS;
+		$is_finished = ( $is_last_batch && $is_success );
+		$new_status  = $is_finished ? 'success' : 'pending';
+
+		if ( $is_success ) {
+			$record->update_meta( 'done', $request['done'] );
+		}
+
+		/**
+		 * Allows filtering the interval between a finished batch process and the closest push of the next one
+		 * from the server.
+		 *
+		 * @since TBD
+		 *
+		 * @param int $interval A value in seconds; defaults to 10.
+		 * @param WP_REST_Request The current batch import request.
+		 */
+		$interval = apply_filters( 'event_aggregator_event_batch_interval', $this->interval_min, $request );
+
+		$interval = is_numeric( $interval ) && (int) $interval > $this->interval_min
+			? min( $this->interval_max, (int) $interval )
+			: $this->interval_min;
+
+		$response_data = array(
+			'status' => 'success',
+			'activity' => $activity->get_items(),
+			'interval' => $interval,
+		);
+
+		if ( $is_finished ) {
+			$record->delete_meta( 'next_batch_hash' );
+		} else {
+			$next_batch_hash = $record->generate_next_batch_hash();
+			$record->update_meta( 'next_batch_hash', $next_batch_hash );
+			$response_data['next_batch_hash'] = $next_batch_hash;
+		}
+
+		$record->set_status( $new_status );
+
+		return new WP_REST_Response( $response_data, 200 );
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function CREATE_args() {
+		return array(
+			'import_id' => array(
+				'required' => true,
+				'type' => 'string',
+				'validate_callback' => array( $this, 'is_valid_import_id' ),
+				'description' => __( 'The import unique ID as provided by Event Aggregator service', 'the-events-calendar' ),
+			),
+			'batch_hash' => array(
+				'required' => true,
+				'type' => 'string',
+				'validate_callback' => array( $this, 'is_expected_batch_hash' ),
+				'description' => __( 'The hash of the next expected batch, as previously provided by the client', 'the-events-calendar' ),
+			),
+			'status' => array(
+				'required' => true,
+				'type' => 'object',
+				'validate_callback' => array( $this, 'is_valid_status_information' ),
+				'description' => __( 'The current status of the import.', 'the-events-calendar' ),
+			),
+			'done' => array(
```

Comment:
`done` seems like a pretty generic word that could hold a number _or_ a boolean. I wonder if changing it to something more explicit would be useful? Like `percentage_complete`, perhaps?